indexing-service/src/main/java/org/apache/druid/indexing/common/task/Task.java

```diff
@@ -164,7 +164,10 @@ default int getPriority()
    * Asks a task to arrange for its "run" method to exit promptly. Tasks that take too long to stop gracefully will be
    * terminated with extreme prejudice.
    *
-   * This method can be called at any time while {@link #run} is being called when the task is killed.
+   * This method can be called at any time while {@link #run} is being called when the task is killed. If this task
+   * is not started yet, that is {@link #run} is not called yet, this method will be never called.
```

Comment:
This text sort of doesn't make sense, because the words like "yet", "before", "after" are not definable in terms of JMM which doesn't deal with time. Yet, in practical terms, currently, `stopGracefully()` *can* be called "before" `run()` in `CompactionTask` on one of the eachSpecs. So this contract is violated.

Reply:
Good point. I think the contract now should say "this method can be called at any time no matter when run() is executed".

> This text sort of doesn't make sense, because the words like "yet", "before", "after" are not definable in terms of JMM which doesn't deal with time.

I don't know what it should be to address [your comment](https://github.com/apache/incubator-druid/pull/8236#discussion_r312195277). What is your suggestion?

Reply:
> I think the contract now should say "this method can be called at any time no matter when run() is executed".

I agree.

> I don't know what it should be to address [your comment](https://github.com/apache/incubator-druid/pull/8236#discussion_r312195277). What is your suggestion?

I would just put the phrase "Regardless when `stopGracefully()` is called w. r. t. `run()`, the implementation must not allow a resource leak or lingering executions (local or remote)."

Reply:
Thanks, added.

Reply:
Missed the first sentence: "this method can be called at any time no matter when run() is executed".

Other than that, I don't have more comments about this PR. I didn't review it in full but don't block it.

Reply:
Thanks, added.